File: stream-impl/src/main/java/org/dustin/kotlin/stream/impl/StreamModule.java
package org.dustin.kotlin.stream.impl;

import org.dustin.kotlin.hello.api.HelloService;
import org.dustin.kotlin.stream.api.StreamService;

import com.google.inject.AbstractModule;
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport;

/**
 * The module that binds the StreamService so that it can be served.
 */
public class StreamModule extends AbstractModule implements ServiceGuiceSupport {
  @Override
  protected void configure() {
    // Bind the StreamService service
    bindService(StreamService.class, StreamServiceImpl.class);
    // Bind the HelloService client
    bindClient(HelloService.class);
    // Bind the subscriber eagerly to ensure it starts up
    bind(StreamSubscriber.class).asEagerSingleton();
  }
}

File: stream-impl/src/main/kotlin/org/dustin/kotlin/stream/impl/StreamModule.kt
package org.dustin.kotlin.stream.impl

import com.google.inject.AbstractModule
import com.lightbend.lagom.javadsl.server.ServiceGuiceSupport
import org.dustin.kotlin.hello.api.HelloService
import org.dustin.kotlin.stream.api.StreamService


class StreamModule : AbstractModule(), ServiceGuiceSupport {
    override fun configure() {
        // Bind the StreamService service
        bindService(StreamService::class.java, StreamServiceImpl::class.java)
        // Bind the HelloService client
        bindClient(HelloService::class.java)
        // Bind the subscriber eagerly to ensure it starts up
        bind(StreamSubscriber::class.java).asEagerSingleton()
    }
}
